src/components/tabBar/index.tsx

```diff
@@ -172,154 +195,103 @@ class TabBar extends Component<TabBarProps, State> {
-    }).start();
-  };
-
-  onScroll = (event: NativeSyntheticEvent<NativeScrollEvent>) => {
-    const {layoutMeasurement, contentOffset, contentSize} = event.nativeEvent;
+  onScroll = event => {
+    const {contentOffset} = event.nativeEvent;
     this.contentOffset = contentOffset;
-    const x = contentOffset.x;
-    const contentWidth = contentSize.width;
-    const containerWidth = layoutMeasurement.width;
-
-    this.animateGradientOpacity(x, contentWidth, containerWidth);
   };
 
   onContentSizeChange = (width: number) => {
     if (this.scrollContentWidth !== width) {
-      const {minTabsForScroll} = this.props;
-      const minChildrenCount = minTabsForScroll || MIN_TABS_FOR_SCROLL;
-
       this.scrollContentWidth = width;
-
+      const {minTabsForScroll} = this.getThemeProps();
+      const minChildrenCount = minTabsForScroll || MIN_TABS_FOR_SCROLL;
       if (this.hasOverflow() && this.childrenCount > minChildrenCount) {
-        this.setState({gradientOpacity: new Animated.Value(1), scrollEnabled: true});
+        this.setState({scrollEnabled: true});
       }
     }
   };
 
-  hasOverflow() {
-    if (this.scrollContentWidth) {
-      if (this.scrollContentWidth > this.scrollContainerWidth) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  renderGradient(height: number, tintColor: string) {
-    const width = GRADIENT_WIDTH;
-
-    if (this.hasOverflow()) {
-      return (
-        <Animated.View
-          pointerEvents="none"
-          style={{
-            width,
-            height: height - INDICATOR_HEIGHT,
-            position: 'absolute',
-            right: 0,
-            opacity: this.state.gradientOpacity
-          }}
-        >
-          <Image
-            source={gradientImage()}
-            style={{width, height: height - INDICATOR_HEIGHT, tintColor}}
-            resizeMode={'stretch'}
-            supportRTL
-          />
-        </Animated.View>
-      );
-    }
-  }
-
   renderTabBar() {
-    const {height, backgroundColor} = this.props;
+    const {height, backgroundColor, containerView, containerProps, gradientMargins} = this.getThemeProps();
     const {scrollEnabled} = this.state;
     const containerHeight = height || DEFAULT_HEIGHT;
 
     return (
-      <View row>
-        <ScrollView
-          ref={r => (this.scrollView = r)}
-          style={{height: containerHeight}}
-          horizontal
-          showsHorizontalScrollIndicator={false}
-          scrollEnabled={scrollEnabled}
-          scrollEventThrottle={16}
-          onScroll={this.onScroll}
-          onContentSizeChange={this.onContentSizeChange}
-          contentContainerStyle={{minWidth: '100%'}}
-        >
-          <View row style={this.styles.tabBar}>
-            {this.renderChildren()}
-          </View>
-        </ScrollView>
-        {this.renderGradient(containerHeight, backgroundColor)}
-      </View>
+      <ScrollBar
+        ref={this.scrollBar}
+        contentContainerStyle={{minWidth: '100%'}}
```

Comment:
move to `styles` otherwise it might cause for unwanted renders..

Reply:
Done